src/python/pants/option/global_options.py

```diff
@@ -15,6 +16,12 @@
 from pants.option.scope import ScopeInfo
 
 
+def _isatty(file_like):
+  # Guard against exotic file-like objects that don't support isatty. Assume those are not TTYs.
+  # This may be an incorrect guess, but it is at least a safe one.
+  return hasattr(file_like, 'isatty') and file_like.isatty()
```

Comment:
could probably just implement `_StreamHandler.isatty()` with a static return value of `False` instead of the checking here - as I think `sys.stdout.isatty()` should ideal always be a safe call to make.

`_StreamHandler` is only used in the daemon context for redirecting daemon-side stdio to the pantsd log, which should never be colorized. `sys.stdout` is then replaced with a `NailgunStreamWriter` when a run is forked which will respect the calling nailgun client's `isatty` ability.